src/Feature.tsx

```diff
@@ -14,157 +14,37 @@
  * limitations under the License.
  */
 import * as React from 'react'
-import { withOptimizely, WithOptimizelyProps } from './withOptimizely'
-import { VariableValuesObject, OnReadyResult, DEFAULT_ON_READY_TIMEOUT } from './client'
-import { getLogger } from '@optimizely/js-sdk-logging'
+import { UserAttributes } from '@optimizely/optimizely-sdk';
+
 
-const logger = getLogger('<OptimizelyFeature>')
+import { VariableValuesObject } from './client'
+import { useFeature } from './hooks';
+import { withOptimizely, WithOptimizelyProps } from './withOptimizely'
 
 export interface FeatureProps extends WithOptimizelyProps {
-  // TODO add support for overrideUserId
   feature: string
   timeout?: number
   autoUpdate?: boolean
+  overrideUserId?: string
+  overrideAttributes?: UserAttributes
   children: (isEnabled: boolean, variables: VariableValuesObject) => React.ReactNode
 }
 
-export interface FeatureState {
-  canRender: boolean
-  isEnabled: boolean
-  variables: VariableValuesObject
-}
-
-class FeatureComponent extends React.Component<FeatureProps, FeatureState> {
-  private optimizelyNotificationId?: number
-  private unregisterUserListener: () => void
-  private autoUpdate: boolean = false
-
-  constructor(props: FeatureProps) {
-    super(props)
-
-    this.unregisterUserListener = () => {}
-
-    const { autoUpdate, isServerSide, optimizely, feature } = props
-    this.autoUpdate = !!autoUpdate
-    if (isServerSide) {
-      if (optimizely === null) {
-        throw new Error('optimizely prop must be supplied')
-      }
-      const isEnabled = optimizely.isFeatureEnabled(feature)
-      const variables = optimizely.getFeatureVariables(feature)
-      this.state = {
-        canRender: true,
-        isEnabled,
-        variables,
-      }
-    } else {
-      this.state = {
-        canRender: false,
-        isEnabled: false,
-        variables: {},
-      }
-    }
+const FeatureComponent = (props: FeatureProps): any => {
```

Comment:
Im not sure what Typescript wants for the return type of this functional component.

I tried a couple things but nothing pleased the `withOptimizely` definition of `Component: React.ComponentType<P>,`

@mjc1283 any ideas for how to improve this?

Reply:
Would it be `React.FC`?

Reply:
returning `JSX.Element` instead of `React.ReactNode` for the `children` render-prop, did the trick for me locally.

```js
export interface FeatureProps extends WithOptimizelyProps {
  feature: string
  timeout?: number
  autoUpdate?: boolean
  overrideUserId?: string
  overrideAttributes?: UserAttributes
  children: (isEnabled: boolean, variables: VariableValuesObject) => JSX.Element
}

const FeatureComponent: React.FC<FeatureProps> = (props: FeatureProps) => {
...
```

I'd have to do some research on the exact differences

Reply:
Paul the wizard!

Reply:
> returning `JSX.Element` instead of `React.ReactNode` for the `children` render-prop, did the trick for me locally.

I see that works, but I think it's semantically incorrect. `props.children` can return any valid react markup, for which `ReactNode` is correct (Doesn't _have_ to be JSX):

`type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;`

I confirmed that the issue is with the `children` prop's return value, specifically.

Upon searching a little bit, it looks like I'm not the only one!

https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18051

The suggestion to wrap the return value in a fragment works (`<>{children(...)}</>`), but to me feels incorrect/clunky, as it's technically modifying the react tree.

What should we do? I suppose returning the children inside a fragment isnt the worst, but I might prefer to tsignore this.

Reply:
I think my vote is to use `any` to get around this and leave a note:

`const FeatureComponent: React.FunctionComponent<FeatureProps> = (props): any => {`

Reply:
@jamesopti Does the fragment workaround give better type safety than using `any`?

Reply:
Using `any` works for me, until this is fixed upstream here https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18051.

If we don't want to wait and prefer better type safety we could change the withOptimizely signature

Reply:
> @jamesopti Does the fragment workaround give better type safety than using any?

It gives better type safety, but at the expense of changing the actual compiled code (to wrap everything in a fragment).

Considering all were doing is rendering children passed by the consumer, I feel like the type safety we gain is small (the consumers `children` function returns valid react code).

But maybe that is useful? I guess injecting a Fragment into the rendered tree isnt so bad either.

Reply:
I went with the Fragment. Decided its the slightly lesser of 2 evils here.